src/chapters/shell-intro/walkthrough_bash-basics.rst

```diff
@@ -0,0 +1,1043 @@
+
+Consider a more complicated example to understand how it works:
+
+.. sourcecode:: bash
+   :caption: Linux/BASH
+
+   $ command $(sub-command $(sub-sub-command))
+
+This command would be evaluated in the following order:
+
+#. innermost level: ``$(sub-sub-command)``
+#. next level: ``$(sub-command <output of sub-sub-command>)``
+#. outermost level: ``command <output of sub-command>``
+
+This is particularly useful in scripts when you want to capture the output of a command in a variable. Because in-line evaluation is a more advanced topic we will return to it later in a context that necessitates it. For now consider the following contrived example where we store our "history" of working directories (WD) in variables to navigate around.
+
+In your ``bash-examples`` directory create another file called ``command-substitution.sh`` and paste in the following contents. We will use the ``echo`` command to print out our CWD throughout the script:
+
+.. sourcecode:: bash
+   :caption: ~/bash-examples/command-substitution.sh
+
+   # in-line evaluation in a string message
+   echo "CWD is: $(pwd)"
+
+   # in-line evaluation to assign the value
+   first_wd=$(pwd)
+
+   cd /tmp
+   echo "CWD is: $(pwd)"
+
+   second_wd=$(pwd)
+
+   cd /usr/bin
+   echo "CWD is: $(pwd)"
+
+   third_wd=$(pwd)
+
+   # return to the first
+   echo "returning to first WD"
+   cd "$first_wd"
+   echo "CWD is: $(pwd)"
+
+   # jump to the second
+   echo "jumping to second WD"
+   cd "$second_wd"
+   echo "CWD is: $(pwd)"
+
+Then execute the script the same way as before:
+
+.. sourcecode:: bash
+   :caption: Linux/BASH
+
+   $ bash ~/bash-examples/command-substitution.sh
+
+.. admonition:: note
+
+   As a good programmer you are likely miffed by the copy and pasting of an identical statement. Although we won't be getting into BASH functions you should be able to make sense of it. Here is a cleaner approach to help you sleep at night!
+
+   .. sourcecode:: bash
+      :caption: ~/bash-examples/command-substitution.sh
+   
+      print_cwd() {
+         echo "CWD is: $(pwd)"
+      }
+
+      print_cwd
+
+      first_wd=$(pwd)
+
+      cd /tmp
+      print_cwd
+
+      second_wd=$(pwd)
+
+      cd /usr/bin
+      print_cwd
+
+      third_wd=$(pwd)
+
+      # return to the first
+      echo "returning to first WD"
+      cd "$first_wd"
+      print_cwd
+
+      # jump to the second
+      echo "jumping to second WD"
+      cd "$second_wd"
+      print_cwd
+
+Learn More
+==========
+
+This has been an introduction to the practical fundamentals of BASH. You are *not expected to have memorized all of it* by any means. Feel free to refer back to this article throughout the course to refresh your memory.
```

Comment:
Change "You are *not expected to have memorized all of it* by any means." to "Learning BASH takes a lot of time. We covered a lot of ground today and will be revisiting the fundamentals regularly until they become second nature." The sentiment is "It's ok if this is a lot, you will get a lot of practice, and you can always look at resources like this", I wouldn't mention anything about having any part of this memorized which most of them will not have retained many of the practicals, or concepts because it is such a big system that is foreign to most if not all of them.

>